cmd/thanos/receive.go

```diff
@@ -489,20 +497,27 @@ func runReceive(
 	}
 
 	if upload {
-		level.Debug(logger).Log("msg", "upload enabled")
-		if err := dbs.Sync(context.Background()); err != nil {
-			level.Warn(logger).Log("msg", "initial upload failed", "err", err)
+		logger := log.With(logger, "component", "uploader")
+		{
+			level.Info(logger).Log("msg", "upload enabled, starting initial sync")
+			ctx, cancel := context.WithTimeout(context.Background(), uploadTimeout)
+			if err := dbs.Sync(ctx); err != nil {
+				cancel()
+				level.Warn(logger).Log("msg", "initial upload failed", "err", err)
```

Comment:
Curious if this should not trigger crash :thinking:

Reply:
Actually good point. We can just stop if we can't upload it.